docker/auth/auth.py

```diff
@@ -189,12 +189,17 @@ def parse_auth(entries, raise_on_error=False):
             'Found entry (registry={0}, username={1})'
             .format(repr(registry), repr(username))
         )
-        conf[registry] = {
-            'username': username,
-            'password': password,
-            'email': entry.get('email'),
-            'serveraddress': registry,
-        }
+        if 'identitytoken' in entry:
+            conf[registry] = {
+                'IdentityToken': entry['identitytoken']
+            }
```

Comment:
Yeah, the token holds all the information. I had Matt Bentley test it with
UCP yesterday and he confirmed it works, too.

On Wed, Sep 14, 2016, 8:44 AM Daniel Nephin <notifications@github.com>
wrote:

> In docker/auth/auth.py
> <https://github.com/docker/docker-py/pull/1210#discussion_r78775816>:
>
> > @@ -189,12 +189,17 @@ def parse_auth(entries, raise_on_error=False):
> >              'Found entry (registry={0}, username={1})'
> >              .format(repr(registry), repr(username))
> >          )
> > -        conf[registry] = {
> > -            'username': username,
> > -            'password': password,
> > -            'email': entry.get('email'),
> > -            'serveraddress': registry,
> > -        }
> > +        if 'identitytoken' in entry:
> > +            conf[registry] = {
> > +                'IdentityToken': entry['identitytoken']
> > +            }
>
> It looks like auth is still required otherwise this function returns
> early. Is that expected or could someone has an identitytoken without an
> auth key?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/docker/docker-py/pull/1210/files/be7d0f01844d5c08ee157446ce96f5bc6381507c..4b46b792eb7bfa0d7cc218e103f43b200f7752a9#r78775816>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABCVnBWNUR-2ZV3xd-2LWjpupwpmpkeLks5qqBZTgaJpZM4J8SKp>
> .
>